### PR TITLE
cleanup: cleanup types.go

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**:

This PR **DOES**:
- remove dead code in types.go
- remove unused types in types.go
- fix typos in code comments and add missing code comments
- change go lang variables that should be constants to constants

This PR **DOES NOT**:
- introduce breaking changes or modify any user facing APIs (you can verify this by looking at the generated config/ files, only field descriptions have changed)
- change any code outside of types.go (everything else in the diff is auto-generated)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by nader-ziada** [comment]
looks good

File: api/v1alpha3/types.go [hunk 1]
```diff
@@ -392,6 +269,7 @@ type AzureSharedGalleryImage struct {
 	Version string `json:"version"`
 }
 
+// AvailabilityZone - DEPRECATED: use FailureDomain instead
```
Comment (nader-ziada) [inline api/v1alpha3/types.go:272]: nit: can we use the standard go lang deprecation notice 
```
"Deprecated:" followed by some information about the deprecation
```
source from here https://blog.golang.org/godoc

Comment (CecileRobertMichon) [reply api/v1alpha3/types.go]: Done, followed the examples in https://golang.org/search?q=Deprecated:

**Pull request merged**